pkg/operator/vttablet/pod.go

```diff
@@ -61,6 +61,8 @@ func NewPod(key client.ObjectKey, spec *Spec) *corev1.Pod {
 func UpdatePodInPlace(obj *corev1.Pod, spec *Spec) {
```

Comment:
What do you think about forgoing DRY here and not creating the `updatePodAnnotations` subroutine? I liked the explicitness of the strategy handlers before, but it's mostly up to opinion.

Reply:
It sounds like I'll be removing updatePodAnnotations entirely and reverting back because we only want to apply annotations instantly for non-pod annotations.

Reply:
It's been removed entirely, so resolving this conversation.